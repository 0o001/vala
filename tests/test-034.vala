using GLib;

class Maman.A : Object {
}

class Maman.B : A {
}

class Maman.C : Object {
}

class Maman.Bar : Object {
	static void main () {
		var a = new A ();
		var b = new B ();
		var c = new C ();

		stdout.printf ("init: %d %d %d\n", null != a, null != b, null != c);

		stdout.printf ("is A: %d %d %d\n", a is A, b is A, c is A);
		stdout.printf ("is B: %d %d %d\n", a is B, b is B, c is B);
		stdout.printf ("is C: %d %d %d\n", a is C, b is C, c is C);

		stdout.printf ("as A: %d %d %d\n", null != (a as A), null != (b as A), null != (c as A));
		stdout.printf ("as B: %d %d %d\n", null != (a as B), null != (b as B), null != (c as B));
		stdout.printf ("as C: %d %d %d\n", null != (a as C), null != (b as C), null != (c as C));
	}
}
