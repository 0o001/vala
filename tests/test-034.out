init: 1 1 1
is A: 1 1 0
is B: 0 1 0
is C: 0 0 1
as A: 1 1 0
as B: 0 1 0
as C: 0 0 1
